persistence/trees/trees.go

```diff
@@ -251,6 +272,66 @@ func (t *treeStore) getStateHash() string {
 	return hexHash
 }
 
+////////////////////////////////
+// AtomicStore Implementation //
+////////////////////////////////
+
+// Savepoint generates a new savepoint for the tree store and saves it internally.
+func (t *treeStore) Savepoint() error {
+	w, err := t.save()
+	if err != nil {
+		return err
+	}
+	t.Prev = w
+	return nil
+}
+
+// Rollback intentionally can't return an error because at this point we're out of tricks
```

Comment:
```suggestion
// Rollback rolls back to the last saved world state maintained by the treeStore.
// It intentionally can't return an error because at this point we're out of tricks
```